include/numerics/tensor_tools.h

```diff
@@ -306,6 +307,56 @@ Number div_from_grad( const TensorValue<Number> & grad );
   divergence of a tensor given the gradient of that tensor. */
 VectorValue<Number> div_from_grad( const TypeNTensor<3,Number> & grad );
 
+template <typename T>
```

Comment:
Could we get a comment here?  I assume this is to distinguish our math classes to allow template specialization based on them, but that's not obvious from the name.